csharp/src/Apache.Arrow/Schema.cs

```diff
@@ -80,15 +79,20 @@ public int GetFieldIndex(string name, StringComparer comparer)
 
         public int GetFieldIndex(string name, IEqualityComparer<string> comparer = default)
         {
-            comparer ??= StringComparer.CurrentCulture;
+            if (comparer == null || comparer.Equals(StringComparer.CurrentCulture))
+            {
+                return _fieldsIndexLookup[name].First();
+            }
 
-            for (int i = 0; i < _fieldsList.Count; i++)
+            for (var i = 0; i < _fieldsList.Count; ++i)
             {
                 if (comparer.Equals(_fieldsList[i].Name, name))
+                {
                     return i;
+                }
             }
 
-            return -1;
+            throw new InvalidOperationException();
```

Comment:
This is a breaking change. Is there a specific reason for it?

Reply:
This is actually fixing a regression that was added in https://github.com/apache/arrow/pull/44576

The function would previously throw `InvalidOperationException` due to `.First(...)`.

I noticed this because I wrote my unit tests before that PR was merged. Are you happy with keeping the `return -1` regression?